src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/FileUtilities.cs

```diff
@@ -1,13 +1,52 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Globalization;
 using System.IO;
 
 namespace Microsoft.AspNetCore.Razor.Utilities;
 
 internal static class FileUtilities
 {
+    private const string RazorExtension = ".razor";
+    private const string CSHtmlExtension = ".cshtml";
+
+    public static bool IsAnyRazorFilePath(ReadOnlySpan<char> filePath, StringComparison comparison)
```

Comment:
It sems that these functions require normalized file paths. It'd probably be good to document that or put it in the parameter name.

Reply:
Not sure I follow this one. When I think "normalized files paths", I think of all slashes facing the same way. This method isn't concerned with slashes at all, except detecting `:/`, and having that also work with `:\` would be wrong, since its only trying to identify documents where the file path is a Uri.

Reply:
When I think of "normalized file paths", I think of Razor's file path normalization which transforms all slashes into forward slashes. So, an absolute Razor-normalized file path on Windows will have a `:/` in it if there's a drive letter. That's what gave me pause here.

Reply:
Thanks, that makes sense. I'll check that my comments are clear enough here. the `:/` we're looking for in this case is in a Uri like `git:/`, not a normalized file path. In fact, and I haven't checked, but I am pretty sure this method would only be called in cohosting and the source generator, and the file path normalizer is not in play in those scenarios.